collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -325,7 +327,7 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
 
                 writableDatabase.execSQL(
                     """
-                    CREATE TABLE ${getRowIdTableName(it)} AS SELECT _id FROM $it;
+                    CREATE TABLE ${getRowIdTableName(it)} AS SELECT _id FROM $it ORDER BY _id;
```

Comment:
There is a potential issue with this assumption: if we delete some entities, their IDs might be reused. While this shouldn't occur under normal circumstances, once we reach the maximum possible ID value of 9223372036854775807, IDs will start to be reused. However, this number is large enough that we don't need to worry about it for now I guess. Do you agree @lognaturel?

Reply:
Ahh the table gets recreated every time we add/delete an entity so it won't be and issue.

Reply:
Preserving order is a nice to have more than a must have so I think it's ok to leave untested. It's only a nice to have because the order is not necessarily meaningful currently. Longer-term, we'll need to be able to let users specify a custom sort order.